Add section on less common package maintenance activities

lee-dohm opened the pull request:
I feel we need a section that covers slightly more advanced package ownership scenarios.

/cc @thedaniel

benogle commented:
This looks good to me. I'm :+1:  to merge pending no @thedaniel changes.

Thanks for adding this.

izuzak commented on book/04-behind-atom/sections/07-maintaining-your-packages.asc:
```diff
@@ -0,0 +1,33 @@
+
+Some packages get too big for one person. Sometimes priorities change and someone else wants to help out. You can let others help or create co-owners by https://help.github.com/articles/adding-collaborators-to-a-personal-repository/[adding them as a collaborator] on the GitHub repository for your package. *Note:* Anyone that has push access to your repository will have the ability to publish new  ...
+
+You can also have packages that are owned by a https://help.github.com/articles/creating-a-new-organization-account/[GitHub organization]. Anyone who is a member of the organization that has push access to the repository for the organization's package will be able to publish new versions of the package.
```
> Anyone who is a member of the organization that has push access to the repository for the organization's package will be able to publish new versions of the package.

Since permissions are managed via teams, here's a slightly different version which mentions teams:

> Anyone who is a member of an organization's https://help.github.com/articles/permission-levels-for-an-organization-repository/[team] which has push access to the package's repository will be able to publish new versions of the package.

izuzak commented on book/04-behind-atom/sections/07-maintaining-your-packages.asc:
```diff
@@ -0,0 +1,33 @@
+
+==== Transferring Ownership
+
+You may one day want to hand off support of your package to someone else. You can do this by simply https://help.github.com/articles/transferring-a-repository/[transferring the GitHub repository] behind your package to the new owner. *Note:* _This is a permanent change! There is no going back!_
```
> You may one day want to hand off support of your package to someone else. You can do this by simply https://help.github.com/articles/transferring-a-repository/[transferring the GitHub repository] behind your package to the new owner.

A bit shorter version for consideration, which also avoids saying "simply" (which is a good guideline in general):

> If you want to hand off support of your package to someone else, you can do that by https://help.github.com/articles/transferring-a-repository/[transferring the package's repository] to the new owner.

izuzak replied:
Also, I'm not sure if just transferring the repository to someone else is enough. @thedaniel will know for sure.

lee-dohm replied:
Yes, it is :grinning: I already asked him about it and verified it when the language-treetop package was transferred to me. That's what drove the creation of this documentation.

izuzak replied:
Great! :heart:

izuzak commented:
:zap: @lee-dohm -- great to have these docs!

lee-dohm commented:
Any other suggestions or should this be merged as-is?

mnquintana commented on book/04-behind-atom/sections/07-maintaining-your-packages.asc:
```diff
@@ -0,0 +1,33 @@
+
+If you want to hand off support of your package to someone else, you can do that by https://help.github.com/articles/transferring-a-repository/[transferring the package's repository] to the new owner. *Note:* _This is a permanent change! There is no going back!_
+
+==== Unpublish Your Package
```
I think we should add a note in this section about taking care to not delete a repo before unpublishing a package.

lee-dohm replied:
Done.

lee-dohm commented:
Added a warning section for not deleting the repository before unpublishing and converted the other cautionary note into an admonition block to match the style.

thedaniel commented:
ship it!
